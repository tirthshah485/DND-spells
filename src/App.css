.app {
	color: var(--text-primary);
	background-color: var(--background);
	font-size: large;
	font-weight: bold;
	padding-top: 20px;
	padding-bottom: 20px;
	min-height: 100vh;
	transition: all .5s;
  }
  button {
	color: var(--text-primary);
	background-color: var(--background);
	border: 2px var(--text-primary) solid;
	float: right;
	transition: all .5s;
  }

  .footer {
	padding: 0 20px;
	z-index: 100;
	width: 100%;
}