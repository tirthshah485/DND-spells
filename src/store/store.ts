import { configureStore } from "@reduxjs/toolkit";
import { combineReducers } from "redux";
import storage from "redux-persist/lib/storage";
import { persistReducer } from "redux-persist";
import thunk from "redux-thunk";
import favouriteSpellReducer from "../features/spell/spellSlice";

const reducers = combineReducers({
    favouriteSpell: favouriteSpellReducer,
});

const persistConfig = {
    key: "root",
    storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

// Stores state tree of the entire application in local memory
export const store = configureStore({
    reducer: persistedReducer,
    devTools: process.env.NODE_ENV !== "production",
    middleware: [thunk],
});
