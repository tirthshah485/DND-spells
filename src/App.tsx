import { AllSpells } from "./pages/AllSpells";
import { FavouriteSpells } from "./pages/FavSpells";
import { SpellInfo } from "./pages/SpellDetail";
import { Header } from "./components/common/Header";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";


import {
    MDBFooter,
} from 'mdb-react-ui-kit';

import { CloudMoon, CloudMoonFill } from "react-bootstrap-icons";

import "./App.css";

import { useState } from "react";

function App() {

    const [theme, setTheme] = useState('dark')
    const styles: {} = {
        position: "fixed",
        bottom: "0px",
        borderRadius: "0px",
        backgroundColor: '#0a4275'
    };
    const switchTheme = () => {
        const newTheme = theme === 'light' ? 'dark' : 'light';
        setTheme(newTheme)
    }
    return (
        <div className="app" data-theme={theme}>


            <Router basename={process.env.PUBLIC_URL}>
                
                <Header
                    text-align="center"
                    color="dark"
                    light={true}
                    dark={true}
                    full={true}
                    expand="md"
                    container="fluid"
                    fixed="top"
                />

                <Routes>
                    <Route path="/" element={<AllSpells />} />
                    <Route path="/favourite" element={<FavouriteSpells />} />
                    <Route path="/spell/:index" element={<SpellInfo />} />
                </Routes>


                <MDBFooter className='bg-dark text-center text-white footer' style={styles}>
                    <div className='text-center p-3' style={{ backgroundColor: 'rgba(0, 0, 0, 0.2)' }}>
                        <span className='me-3'>The 5th Edition Dungeons and Dragons API</span>
                        <a className='text-white' href='http://www.dnd5eapi.co/'>
                            Know More
                        </a>
                        <button onClick={switchTheme} className="btn btn-secondary btn-sm">

                            Click here to Change Mode   {theme === 'light' ? <CloudMoon size={20} /> : <CloudMoonFill size={20} />}
                        </button>

                    </div>
                </MDBFooter>

            </Router>



        </div>
    );
}

export default App;
