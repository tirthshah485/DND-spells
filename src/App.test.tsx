import React from "react";
import { Provider } from "react-redux";
import { store } from "./store/store";
import App from "./App";
import renderer from "react-test-renderer";

it("Matches App snapshot", () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <App />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
