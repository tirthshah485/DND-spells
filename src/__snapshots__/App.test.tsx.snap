// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches App snapshot 1`] = `
Array [
  <div
    data-testid="header"
  >
    <nav
      className="navbar navbar-expand-md navbar-light navbar-dark bg-dark fixed-top"
      full={true}
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="logo"
            src="./dnd.jpg"
            style={
              Object {
                "height": 40,
                "margin": "0 15px",
                "width": 40,
              }
            }
          />
          DnD Spells
        </a>
        <button
          aria-label="Toggle navigation"
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="navigation ms-auto navbar-nav"
            data-testid="navigation"
          >
            <li
              className="nav-item"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                All Spells
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                href="/favourite"
                onClick={[Function]}
              >
                Favourites
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="main pb-5"
  >
    <h1
      className="display-2 text-center pb-3"
    >
      All Spells
    </h1>
    <div
      className="spell-list px-5 py-2"
    >
      <div
        className="spinner-border-sm spinner-border"
        role="status"
      >
        <span
          className="visually-hidden"
        >
          Loading...
        </span>
      </div>
      <span
        className="spell-loading"
        data-testid="loading"
      >
        Loading...
      </span>
    </div>
  </div>,
  <div
    className="footer card bg-dark"
    data-testid="footer"
    style={
      Object {
        "borderRadius": "0px",
        "bottom": "0px",
        "position": "fixed",
      }
    }
  >
    <div
      className="img-container"
    >
      <img
        alt="Card image cap"
        className="card-img-bottom"
        src="./dnd-logo-footer.png"
      />
    </div>
  </div>,
]
`;
